Sources/SwiftFrameCore/Workers/AttributedStringCache.swift

```diff
@@ -0,0 +1,88 @@
+import AppKit
+import Foundation
+
+// Used for keeping attributed strings that were parsed on the main thread in memory for asynchronous processing
+final class AttributedStringCache: NSObject {
+
+    // MARK: - Nested Type
+
+    enum FontMode {
+        case dynamic(maxSize: CGFloat)
+        case fixed(pointSize: CGFloat)
+    }
+
+    // MARK: - Properties
+
+    static let shared = AttributedStringCache()
+    private lazy var cache = NSCache<NSString, NSAttributedString>()
+
+    // MARK: - Reading/Writing to cache
+
+    func getValue(titleIdentifer: String, locale: String, deviceIdentifier: String) throws -> NSAttributedString {
+        let identifier = makeCacheIdentifier(titleIdentifer: titleIdentifer, locale: locale, deviceIdentifier: deviceIdentifier)
+        guard let value = cache.object(forKey: NSString(string: identifier)) else {
+            throw NSError(description: "Could not find value for key \"\(identifier)\" in attributed string cache")
+        }
+        return value
+    }
+
+    func process(
+        _ associatedStrings: [AssociatedString],
+        locale: String,
+        deviceIdentifier: String,
+        maxFontSizeByGroup: [String: CGFloat],
+        font: NSFont,
+        color: NSColor,
+        maxFontSize: CGFloat
+    ) throws {
+        guard Thread.current.isMainThread else {
+            throw NSError(description: "Attributed string processing has to be done on the main thread")
+        }
+        try associatedStrings.forEach {
```

Comment:
**nitpick** all the `$0` in body are hard to read. Suggestion to use regular for loop with proper variable name